# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/cairo/cairo/src/cairo-pattern-private.h
# Commit: 3cdfb39376b7
# Full Hash: 3cdfb39376b736597fe658a322389117686b7e82
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2024-05-07 09:13:07
# Regressor Bug: 1892913
# File Overlap Count: 3
# Description:
#   Bug 1892913 - patch 1 - Update cairo source to 1.18.0 r=gfx-reviewers,lsalzman
#   
#   Wholesale replacement of files in gfx/cairo/cairo with contents of the 1.18.0 release tarball,
#   omitting subdirectories that are not part of the core library build.
#   
# ==============================================================================

diff -r 9daa351f875b -r 3cdfb39376b7 gfx/cairo/cairo/src/cairo-pattern-private.h
--- a/gfx/cairo/cairo/src/cairo-pattern-private.h	Mon May 06 19:22:35 2024 +0000
+++ b/gfx/cairo/cairo/src/cairo-pattern-private.h	Mon May 06 19:25:12 2024 +0000
@@ -52,6 +52,7 @@
     CAIRO_PATTERN_NOTIFY_FILTER = 0x2,
     CAIRO_PATTERN_NOTIFY_EXTEND = 0x4,
     CAIRO_PATTERN_NOTIFY_OPACITY = 0x9,
+    CAIRO_PATTERN_NOTIFY_DITHER = 0x12,
 };
 
 struct _cairo_pattern_observer {
@@ -72,6 +73,8 @@
     cairo_filter_t		filter;
     cairo_extend_t		extend;
     cairo_bool_t		has_component_alpha;
+    cairo_bool_t		is_foreground_marker;
+    cairo_dither_t		dither;
 
     cairo_matrix_t		matrix;
     double			opacity;
@@ -86,6 +89,12 @@
     cairo_pattern_t base;
 
     cairo_surface_t *surface;
+
+    /* This field is only used by the wrapper surface for retreiving
+     * the region id from the target during create regions and passing
+     * the region id to the target surface during playback.
+     */
+    unsigned int region_array_id;
 } cairo_surface_pattern_t;
 
 typedef struct _cairo_gradient_stop {
@@ -233,6 +242,9 @@
 cairo_private cairo_pattern_t *
 _cairo_pattern_create_solid (const cairo_color_t	*color);
 
+cairo_private cairo_pattern_t *
+_cairo_pattern_create_foreground_marker (void);
+
 cairo_private void
 _cairo_pattern_transform (cairo_pattern_t      *pattern,
 			  const cairo_matrix_t *ctm_inverse);
@@ -308,15 +320,15 @@
 _cairo_pattern_get_ink_extents (const cairo_pattern_t	    *pattern,
 				cairo_rectangle_int_t       *extents);
 
-cairo_private unsigned long
+cairo_private uintptr_t
 _cairo_pattern_hash (const cairo_pattern_t *pattern);
 
-cairo_private unsigned long
-_cairo_linear_pattern_hash (unsigned long hash,
+cairo_private uintptr_t
+_cairo_linear_pattern_hash (uintptr_t hash,
 			    const cairo_linear_pattern_t *linear);
 
-cairo_private unsigned long
-_cairo_radial_pattern_hash (unsigned long hash,
+cairo_private uintptr_t
+_cairo_radial_pattern_hash (uintptr_t hash,
 			    const cairo_radial_pattern_t *radial);
 
 cairo_private cairo_bool_t